src/block_cache/block_chunkcache.c

```diff
@@ -554,20 +609,31 @@ __wt_chunkcache_setup(WT_SESSION_IMPL *session, const char *cfg[], bool reconfig
           session, &chunkcache->hashtable[i].bucket_lock, "chunk cache bucket lock"));
     }
 
-    if (chunkcache->type != WT_CHUNKCACHE_IN_VOLATILE_MEMORY) {
-#ifdef ENABLE_MEMKIND
-        WT_RET(memkind_create_pmem(chunkcache->dev_path, 0, &chunkcache->memkind));
-#else
-        WT_RET_MSG(session, EINVAL, "Chunk cache that is not in DRAM requires libmemkind");
-#endif
-    }
-
     WT_RET(__wt_thread_create(
       session, &evict_thread_tid, __chunkcache_eviction_thread, (void *)session));
 
     chunkcache->configured = true;
     __wt_verbose(session, WT_VERB_CHUNKCACHE, "configured cache in %s, with capacity %" PRIu64 "",
       (chunkcache->type == WT_CHUNKCACHE_IN_VOLATILE_MEMORY) ? "volatile memory" : "file system",
       chunkcache->capacity);
+
+    return (0);
+}
+
+/*
+ * __wt_chunkcache_teardown --
+ *     Tear down the chunk cache.
+ */
+int
+__wt_chunkcache_teardown(WT_SESSION_IMPL *session)
+{
+    WT_CHUNKCACHE *chunkcache;
+
+    chunkcache = &S2C(session)->chunkcache;
+    if (!chunkcache->configured || chunkcache->type == WT_CHUNKCACHE_IN_VOLATILE_MEMORY)
```

Comment:
We seem to have a boolean and a flag to represent if the chunkcache is configured. For the sake of consistency, we should use one or the other. I believe that we should use the flag and get rid of the bool.
`
 if (chunkcache->configured !=  WT_CHUNKCACHE_CONFIGURED || chunkcache->type == WT_CHUNKCACHE_IN_VOLATILE_MEMORY)`

Reply:
We can take care of this in later tickets.